network-area-diagram/src/main/java/com/powsybl/nad/build/iidm/NetworkGraphBuilder.java

```diff
@@ -203,19 +203,19 @@ private VoltageLevelNode getVoltageLevelNode(Graph graph, Terminal terminal) {
                 .orElseThrow(() -> new PowsyblException("Cannot add edge, corresponding voltage level is unknown: '" + terminal.getVoltageLevel().getId() + "'"));
     }
 
-    private ThreeWindingsTransformer.Side[] getSidesArray(ThreeWindingsTransformer.Side sideA) {
-        ThreeWindingsTransformer.Side sideB;
-        ThreeWindingsTransformer.Side sideC;
-        if (sideA == ThreeWindingsTransformer.Side.ONE) {
-            sideB = ThreeWindingsTransformer.Side.TWO;
-            sideC = ThreeWindingsTransformer.Side.THREE;
-        } else if (sideA == ThreeWindingsTransformer.Side.TWO) {
-            sideB = ThreeWindingsTransformer.Side.ONE;
-            sideC = ThreeWindingsTransformer.Side.THREE;
+    private ThreeSides[] getSidesArray(ThreeSides sideA) {
+        ThreeSides sideB;
+        ThreeSides sideC;
+        if (sideA == ThreeSides.ONE) {
+            sideB = ThreeSides.TWO;
+            sideC = ThreeSides.THREE;
+        } else if (sideA == ThreeSides.TWO) {
+            sideB = ThreeSides.ONE;
+            sideC = ThreeSides.THREE;
         } else {
-            sideB = ThreeWindingsTransformer.Side.ONE;
-            sideC = ThreeWindingsTransformer.Side.TWO;
+            sideB = ThreeSides.ONE;
+            sideC = ThreeSides.TWO;
         }
-        return new ThreeWindingsTransformer.Side[] {sideA, sideB, sideC};
+        return new ThreeSides[] {sideA, sideB, sideC};
     }
```

Comment:
Similarly, we don't have full coverage here, I think you can rewrite this with `ThreeSides::getNum` and `ThreeSides::valueOf` with a lot less lines.